### PR TITLE
Video playback: remove chunking logic

### LINKED ISSUE #2783: [Bug] DASH playback does not work on Crystal 1.3.0
**Describe the bug**
Playing any video with DASH on an Invidious instance compiled with Crystal 1.3.0 produces no playback a flood of the following warnings:
```
VIDEOJS: WARN: Problem encountered with playlist 0-placeholder-uri-0. HLS request errored at URL: http://localhost/videoplayback?expire=...&ei=...&host=... Switching to playlist 6-placeholder-uri-6.
```
```
VIDEOJS: WARN: Problem encountered with playlist 5-placeholder-uri-5. Problem encountered loading the default audio track. Switching to playlist 4-placeholder-uri-4.
```
(with numbers around `-placeholder-uri-` being different every time)
Non-DASH playback works fine, regardless of whether or not it is being proxied.

**Steps to Reproduce**
1. Install Crystal 1.3.0
2. Rebuild Invidious
3. Attempt playing any video with DASH (audio-only playback seems unaffected)

**Logs**
I've looked through the logs, but was unable to find anything beyond logged HTTP requests. I can append the full logs if you're unable to reproduce the issue.

**Screenshots**
N/A

**Additional context**
System: Debian 11/stable/Bullseye, Linux 5.10.0-8, x86_64
Crystal version: 1.3.0
LLVM version: 10.0.0
Invidious version: seems irrelevant, I've tried a few commits between f54e247eb421795f7d7351516978e0f25e8a14f5 and 5a728243a2c3f12d80d08e81fc7d424889b55394, and they've all had the same issue

### PR DESCRIPTION
**Pull request by SamantazFox**
Fixes #2783
Can probably fix a few more issues (TBD)

This also improves video load time quite a lot.

So far, it seems to work fine with both HTTP and QUIC clients.

I've tested the following:
 - [x] DASH playback
 - [x] Normal playback (HD720)
 - [x] Downloads
 - [x] "listen" mode

---

Note 1: this has to be extensively tested, as this is quite a breaking change.

Note 2: deployed on https://test.invidious.io

### CODE REVIEW

**Comment by matthewmcgarvey** [comment]
I was curious about this reason for the chunk code in the first place so I looked into the history for where that code was added. I found it and the commit message is `Chunk video files to bypass throttling`. Did you know the chunk code was for that and if it matters anymore to remove it?

https://github.com/iv-org/invidious/commit/e5fa5df7be38d60ce9e4835bae170d805482afc0

File: src/invidious.cr [hunk 1]
```diff
@@ -51,9 +51,23 @@ CHARS_SAFE         = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz012345
...
+  "cache-control",
+  "content-length",
+  "if-none-match",
+  "range",
+}
+RESPONSE_HEADERS_BLACKLIST = {
+  "access-control-allow-origin",
+  "alt-svc",
+  "server",
+  "content-security-polic",
```
[the remark this answers is not in the document]
Comment (SamantazFox) [reply src/invidious.cr:66]: That's a typo ^^

**Comment by unixfox** [comment]
> I was curious about this reason for the chunk code in the first place so I looked into the history for where that code was added. I found it and the commit message is `Chunk video files to bypass throttling`. Did you know the chunk code was for that and if it matters anymore to remove it?
> 
> [e5fa5df](https://github.com/iv-org/invidious/commit/e5fa5df7be38d60ce9e4835bae170d805482afc0)

A lot has changed over the years, especially now that we fetch differently the way of getting the video streams, so I guess this code is not relevant anymore and can be removed.

**Comment by SamantazFox** [comment]
> I was curious about this reason for the chunk code in the first place so I looked into the history for where that code was added. I found it and the commit message is `Chunk video files to bypass throttling`. Did you know the chunk code was for that [snip]

If I'm not mistaken, the chunking code almost never gets used, as most requests are far from exceeding that `HTTP_CHUNK_SIZE` limit. And yes, I'm aware that is was some throttling mitigation, however it was causing issues with crystal 1.3.x (I'm still not sure why. I'm guessing that this is an obscure conflict between our manual chunking and how the stdlib handles headers).

> and if it matters anymore to remove it?

As said, it solves the compatibility with 1.3.x, but that could be a breaking change (see "Note 1"). If removing that code breaks too much things, what we could do is to split the requested range into smaller requests and parallelize them using fibers.

**Review by matthewmcgarvey — APPROVED** [review]
Code itself looks fine, but it will be up to y'alls testing to make sure it is correct. I don't know a thing about this flow

**Comment by SamantazFox** [comment]
Ok, seems like normal (non-dash) video has issues (this is only visible on long videos).

**Comment by SamantazFox** [comment]
Useless, now that we know that it's caused by kemal.

**Pull request closed without merging**